src/Cron.php

```diff
@@ -131,4 +156,47 @@ public static function isValid(string $expression): bool
     {
         return CronExpression::isValidExpression($expression);
     }
+
+    public static function fromJson(string $data): Frequency
+    {
+        $data = json_decode($data, true);
+        $start = null;
+        $end = null;
+
+        if (isset($data['start'])) {
+            $start = new DateTime($data['start']);
+            unset($data['start']);
+        }
+
+        if (isset($data['end'])) {
+            $end = new DateTime($data['end']);
+            unset($data['end']);
+        }
+
+        $self = new static(current($data));
```

Comment:
I find it odd that you rely on the cursor position here. Why not just use a key like for end and start as well?